pkg/controller/v1beta1/inferenceservice/components/explainer.go

```diff
@@ -106,6 +109,12 @@ func (e *Explainer) Reconcile(isvc *v1beta1.InferenceService) error {
 		if err := controllerutil.SetControllerReference(isvc, r.Service.Service, e.scheme); err != nil {
 			return errors.Wrapf(err, "fails to set service owner reference for explainer")
 		}
+		//set autoscaler Controller
+		if r.Scaler.Autoscaler.AutoscalerClass == constants.AutoscalerClassHPA {
+			if err := controllerutil.SetControllerReference(isvc, r.Scaler.Autoscaler.HPA.HPA, e.scheme); err != nil {
```

Comment:
can `r.Scaler.Autoscaler.HPA.HPA` be nil here ?

Reply:
no,  if user doesn't set the hpa, kserve will create a hpa by default.

Reply:
right, I see there is still chance that createHPA returns nil if metrics annotation is not valid which could crash the controller

https://github.com/kserve/kserve/pull/1830/files#diff-6ca8bcf77ba1224c3b3978f9de1fb5c00d62b181d9e1db7c48700cb671356499R100

Reply:
If createHPA failed, this will make createAutoscaler failed, and make rawkubereconciler failed. so this will not happen at this section.